feat: external link and composer pages extensions docs (#1711)
mia-platform/documentation: docs/console/company-configuration/extensions.md

sidebar_label: Manage Extensions
---

> Extensions allow expanding Console capabilities by adding custom pages to the sidebar navigation, to provide users with new pages
> tailored to your Company needs or to integrate external tools with a seamless experience within the product.

Reviewer: ```suggestion
Extensions allow expanding Console capabilities by adding custom pages to the sidebar navigation. This allows you to provide users with new pages tailored to your Company's needs or integrate external tools with a seamless experience within the product.
```

## Extension Manager

Extensions can be managed through a dedicated section called **Extensibility** placed in the Company Overview page.

:::info
Only **Company Owners** have the necessary permissions to see and use this section.
Moreover, extensions are **only accessible within the Company** they are registered into.
:::

![extension placeholder](./img/extensions/extension-placeholder.png)

There are different extension types supported:

- iframe: allows you to embed web pages in Console using an `<iframe>` tag
- External link: these extensions will be shown as menu items in the sidebar but opened in a new window on click

:::tip
Embedding a web app as an *iframe* extension may require additional configurations,
such as making sure `x-frame-options` header or the `frame-ancestors` Content Security Policy are properly configured
:::

## Add new Extension

> In any moment, you can add a new extensions to your Company; creating an extension requires the following configurations to be filled:

Reviewer: ```suggestion
You can add a new extension to your Company at any moment; creating an extension requires the following configurations to be set:
```

- **Extension name** (*required*): the name for your extension

> - **Entry URL** (*required*): the URL of the web page that will load when accessing your extension (This will be embedded inside an iframe or opened in a new window depending on the extension type)

Reviewer: ```suggestion
- **Entry URL** (*required*): the URL of the web page that will be displayed when accessing your extension (This will be embedded inside an iframe or opened in a new window depending on the extension type)
```

> - **Destination Area** (*required*): the destination where your extension menu item will be placed

Reviewer: Here we could explain what these "destinations" are (specific areas of the Console) and mention the currently available destinations

Author: >Here we could explain what these "destinations" are (specific areas of the Console)

on this I agree

> and mention the currently available destinations

I'd avoid this as it could become an outdated list pretty rapidly 🤔

> - **Category** (*required*): the menu category that will contain the extension menu item

Reviewer: ```suggestion
- **Category** (*required*): the menu category that will contain the extension menu item

:::info
The available **Category** options depend on the selected **Destination Area**.
:::
```

> - **Visibility**: contexts in which you want to make your extension visible. If nothing is checked, the extension will be inactive.

Reviewer: ```suggestion
- **Visibility**: You might want to make your extension visible on the whole Company, or maybe just on a few Projects. This field lets you set the contexts in which you want to make your extension visible. If nothing is checked, the extension will be inactive.
```

:::info

> The **Category** options are based on the selected destination area.
> 
> The **Visibility** can be applied only on the *Whole Company* context when the *Company Overview* destination area is selected.

Reviewer: I would put the info related to each field close to where the field is described to avoid confusion

```suggestion
When the *Company Overview* destination area is selected, the only available **Visibility** option will be the *Whole Company*.
```

- [Add new extensions](#add-new-extension)
- [Access the detail page](#extension-detail) of an extension by clicking on its name
- Delete an extension

## Extension Detail

The extension detail page allows you to view the main information that characterizes the registered extension, which can also be modified through the corresponding *Edit* buttons.

![extension detail](./img/extensions/extension-detail.png)

### Extension Info

The *Extension Info* card contains the identifying information for the extension, where only the name and entry URL can be edited.

![edit extension info](./img/extensions/edit-extension-info.png)

### Extension Location

The *Extension Location* card contains the information related to the positioning of the extension menu item within the Console; these details are fully editable.

![edit extension destination](./img/extensions/edit-extension-destination.png)

### Extension Visibility

The *Extension Visibility* card shows in which contexts the extension is active and visible.

![edit extension visibility](./img/extensions/edit-extension-visibility.png)

:::info
Just as during creation, Visibility is tied to the Destination Area set for the extension.

> For example, if the Destination Area is set to *Company Overview*, then visibility can only be applied to the *Whole Company* option.

Reviewer: ```suggestion
For example, if the Destination Area is set to *Company Overview*, the only available visibility option will be *Whole Company*.
```
